File: backend/app/manejadores/GestionUsuario.go
package manejadores

import(
	"mizcuin/modelos"
	"mizcuin/repositorios"
)

func GuardarUsuario(usuario modelos.Usuario) bool{
	consulta := "INSERT INTO Usuario (nombre, apellido, email, telefono, contraseña) VALUES (?, ?, ?, ?, ?)"
	
	return repositorios.Guardar(consulta, usuario)
}

func BuscarUsuario(usuario modelos.Usuario) (modelos.Usuario, bool){
	consulta := "SELECT * FROM Usuario WHERE id_usuario = ?"
	
	return repositorios.BuscarIdUsuario(consulta, usuario.Id_usuario)
}

func ActualizarUsuario(usuario modelos.Usuario) bool{
	consulta := "UPDATE Usuario SET nombre = ?, apellido = ?, email = ?, telefono = ?, contraseña = ? WHERE id_usuario = ?"
	
	return repositorios.Actualizar(consulta, usuario)
}

func EliminarUsuario(usuario modelos.Usuario) bool{
	consulta := "DELETE FROM Usuario WHERE id_usuario = ?"
	
	return repositorios.Eliminar(consulta, usuario)
}

File: backend/app/manejadores/gestionUsuario.py
from repositorios import usuarioRepo
from modelos.modelos import AnimalLover

def guardarUsuario(usuario: AnimalLover) -> bool:
    consulta = "INSERT INTO animalLover (nombre, apellido, email, telefono, contraseña) VALUES (?, ?, ?, ?, ?)"
    return usuarioRepo.guardarUsuario(consulta, usuario)

def buscarUsuario(idUsuario: int) -> tuple[AnimalLover, bool]:
    consulta = "SELECT * FROM animalLover WHERE id_animalLover = ?"
    return usuarioRepo.buscarIdUsuario(consulta, idUsuario)

def actualizarUsuario(usuario: AnimalLover) -> bool:
    consulta = "UPDATE animalLover SET nombre = ?, apellido = ?, email = ?, telefono = ?, contraseña = ? WHERE id_animalLover = ?"
    return usuarioRepo.actualizarUsuario(consulta, usuario)

def eliminarUsuario(idUsuario: int) -> bool:
    consulta = "DELETE FROM animalLover WHERE id_animalLover = ?"
    return usuarioRepo.eliminarUsuario(consulta, idUsuario)
